classes/Context.php

```diff
@@ -370,10 +371,16 @@ public function getTranslator()
             return $this->translator;
         }
 
-        $translator = $this->getTranslatorFromLocale($this->language->locale);
-        $this->translator = $translator;
+        $sfContainer = SymfonyContainer::getInstance();
 
-        return $translator;
+        if (null === $sfContainer) {
+            // symfony's container isn't available in front office, so we load and configure the translator component
+            $this->translator = $this->getTranslatorFromLocale($this->language->locale);
```

Comment:
Can you confirm this method is _NEVER_ called in the BO context? (either legacy controller or symfony controller)

Reply:
Yes, I carefully checked that we never enter this method in BO context, furthermore, I checked that the **TranslatorComponent** class is never instantiated in BO context.

The reason it works like this is that the SymfonyContainer is always available in BO context, in legacy and migrated pages alike.

Reply:
Maybe it would be nice to improve the condition and check for BO context? I'm not sure about the proper way to do that though... ❓ 
I the mean time I like this condition, because internally, it is really the availability or not of the **SymfonyContainer** that implies which Translator will be used.

Reply:
Okay!